_posts/2018-06-18-composing-queries-with-ecto

```diff
@@ -0,0 +1,199 @@
+---
+author: Sean Callan
+author_link: https://github.com/doomspork
+categories: post
+date:   2018-06-20
+layout: post
+title:  Queries composition with Ecto
+excerpt: >
+	Follow along as we look at how to dynamically compose Ecto queries using pattern matching and reduction.
+	
+---
+
+Ecto is fantastic tool that affords us a great degree of flexibility.
+In this blog post we'll look at how we can dynamically build our Ecto queries and sanitize our input data at the same time.
+
+Let's plan to approach our composition in 3 steps:
+
+- Create the base query we'll build upon
+- Compose our query from input criteria
+- Execute our final query
+
+For our example we'll be working with everyone's favorite example project: a blog!
+Before we begin take a peek at the schema we'll be building our code to interface with:
+
+![image](https://user-images.githubusercontent.com/73386/41698787-7a4efb0e-74dd-11e8-970b-7fb8fe3fef14.png)
+
+## Query base
+
+To keep things clean we'll create a new module to contain the functionality for accessing the underlying schema data.
+Let's move ahead with creating our module and addressing the first step above: the base query.
+
+```elixir
+defmodule Posts do
+  import Ecto.Query
+
+  defp base_query do
+    (from p in Post)
+  end
+end  
+```
+
+Simple enough.
+When `base_query/0` is called we'll create the initial query that will serve as the base for our criteria.
+At this point our query is analogous to `SELECT * FROM posts`.
+
+## Applying our criteria
+
+Next we'll need to build upon `base_query/0` by applying our criteria, this is where the magic of our query composition shines!
+
+There's a good chance when we apply our criteria it won't be simple comparisons.
```

Comment:
Can you rephrase this? Maybe this must become `There's a good chance when we apply our criteria, it won't be a simple comparison.`